### PR TITLE
PWGHF: Adding D0 to taskFlowCharmHadrons, including Ml.

### PR DESCRIPTION
**Pull request by Libeiyu**

### CODE REVIEW

File: PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx [hunk 1]
```diff
@@ -150,17 +156,20 @@ struct HfTaskFlowCharmHadrons {
...
-    float phiTrack2 = std::atan2(pYtrack2, pXtrack2);
 
     tracksQx.push_back(cos(harmonic * phiTrack0) * pTtrack0 / ampl);
     tracksQy.push_back(sin(harmonic * phiTrack0) * pTtrack0 / ampl);
     tracksQx.push_back(cos(harmonic * phiTrack1) * pTtrack1 / ampl);
     tracksQy.push_back(sin(harmonic * phiTrack1) * pTtrack1 / ampl);
-    tracksQx.push_back(cos(harmonic * phiTrack2) * pTtrack2 / ampl);
-    tracksQy.push_back(sin(harmonic * phiTrack2) * pTtrack2 / ampl);
+
+    if constexpr (DecayChannel != DecayChannel::D0ToPiK) {
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx:165]: Please avoid this kind of confusing naming patterns. `DecayChannel` on the left is a variable, `DecayChannel` on the right is a type.

File: PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx [hunk 2]
```diff
@@ -150,17 +156,20 @@ struct HfTaskFlowCharmHadrons {
...
+      float pXtrack2 = cand.pxProng2();
+      float pYtrack2 = cand.pyProng2();
+      float pTtrack2 = cand.ptProng2();
+      float phiTrack2 = std::atan2(pYtrack2, pXtrack2);
+      tracksQx.push_back(cos(harmonic * phiTrack2) * pTtrack2 / ampl);
+      tracksQy.push_back(sin(harmonic * phiTrack2) * pTtrack2 / ampl);
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx:171]: Explicitly state the namespace you use.

File: PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx [hunk 3]
```diff
@@ -50,8 +51,9 @@ struct HfTaskFlowCharmHadrons {
   Configurable<float> zVtxMax{"zVtxMax", 10., "Max vertex coordinate z"};
   Configurable<int> harmonic{"harmonic", 2, "harmonic number"};
   Configurable<int> qvecDetector{"qvecDetector", 0, "Detector for Q vector estimation (FV0A: 0, FT0M: 1, FT0A: 2, FT0C: 3, TPC Pos: 4, TPC Neg: 5)"};
-  Configurable<int> centDetector{"centDetector", 0, "Detector for centrality estimation (V0A: 0, T0M: 1, T0A: 2, T0C: 3"};
+  Configurable<int> centDetector{"centDetector", 0, "Detector for centrality estimation (V0A: 0, T0M: 1, T0A: 2, T0C: 3)"};
   Configurable<int> selectionFlag{"selectionFlag", 1, "Selection Flag for hadron (e.g. 1 for skimming, 3 for topo. and kine., 7 for PID)"};
+  Configurable<int> prongNum{"prongNum", 3, "Number of candidate's prong (For D0, set selectionFlag = 1 and prongNum = 2)"};
```
Comment (vkucera) [inline PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx:56]:
```suggestion
  Configurable<int> nProngs{"nProngs", 3, "Number of candidate's prong (For D0, set selectionFlag = 1 and nProngs = 2)"};
```
Why does this have to be set manually?

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx (hunk 1 above)
Comment (Libeiyu) [reply PWGHF/D2H/Tasks/taskFlowCharmHadrons.cxx:165]: Sorry, I have closed this PR due to a failed submission and will resubmit. I'll change or explain existing comments.

**Comment by Libeiyu** [comment]
This is the new PR [](https://github.com/AliceO2Group/O2Physics/pull/5089) @vkucera. Thank you for the comment.

**Comment by vkucera** [comment]
@Libeiyu In general, there is no reason to close a PR and open a new one from the same branch. The new PR will be identical with the closed one.

**Pull request closed without merging**